src/lib.rs

```diff
@@ -118,6 +121,237 @@ pub fn builder() -> Builder {
     Default::default()
 }
 
+/// Construct a new [`Builder`](./struct.Builder.html) from command line flags.
+pub fn builder_from_flags<I>(args: I) -> Result<Builder, io::Error>
+    where I: Iterator<Item=String>
+{
+    // clap doesn't support the end of options marker, --
+    // https://github.com/kbknapp/clap-rs/issues/735
+    let mut terminated = false;
+    let (bindgen_args, clang_args): (Vec<_>, Vec<_>) = args.partition(|arg| {
```

Comment:
Probably if arg was a slice, it'd be easier to do something like `postition()`, and then `split_at`.